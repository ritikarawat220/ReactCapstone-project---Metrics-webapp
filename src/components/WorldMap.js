import { useSelector } from 'react-redux';
import worldMap from '../assets/world.png';
import asia from '../assets/asie.png';
import africa from '../assets/afrique.png';
import europe from '../assets/leurope.png';
import northAmerica from '../assets/amerique-du-nord.png';
import southAmerica from '../assets/amerique-du-sud.png';
import oceania from '../assets/australie.png';
import antarctica from '../assets/antarctica.png';

const WorldMap = () => {
  const state = useSelector((state) => state.countries.map);
  let continent;
  switch (state) {
    case 'Asia':
      continent = <img src={asia} alt="Asian continent" />;
      break;
    case 'Africa':
      continent = <img src={africa} alt="African continent" />;
      break;
    case 'Europe':
      continent = <img src={europe} alt="European continent" />;
      break;
    case 'North America':
      continent = <img src={northAmerica} alt="North american continent" />;
      break;
    case 'South America':
      continent = <img src={southAmerica} alt="South american continent" />;
      break;
    case 'Antarctica':
      continent = <img src={antarctica} alt="Antarctican continent" />;
      break;
    case 'Oceania':
      continent = <img src={oceania} alt="Oceania continent" />;
      break;
    default:
      continent = <img src={worldMap} alt="World Map" />;
      break;
  }

  return <div className="map" data-testid="map-div">{continent}</div>;
};

export default WorldMap;
