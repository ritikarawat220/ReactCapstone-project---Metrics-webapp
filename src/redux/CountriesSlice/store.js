import { configureStore } from '@reduxjs/toolkit';
import countriesSlice from './CountriesSlice';

const store = configureStore({
  reducer: {
    countries: countriesSlice,
  },
});

export default store;
