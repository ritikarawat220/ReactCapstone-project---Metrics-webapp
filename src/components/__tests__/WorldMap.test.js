import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import store from '../../redux/CountriesSlice/store';
import WorldMap from '../WorldMap';

describe('Testing the WorldMap', () => {
  it('The Worldmap component should match the snapshot: ', () => {
    const picker = render(
      <Provider store={store}>
        <WorldMap />
      </Provider>,
    );
    expect(picker).toMatchSnapshot();
  });
});
