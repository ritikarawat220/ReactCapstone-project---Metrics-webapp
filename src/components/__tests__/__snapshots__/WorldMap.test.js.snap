// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the WorldMap The WorldMap component should match the snapshot:  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="map"
        data-testid="map-div"
      >
        <img
          alt="flat texture blue green World transparent background @transparentpng.com"
          class="world-image"
          src="https://upload.wikimedia.org/wikipedia/commons/a/a9/Rotating_earth_%28large%29_transparent.gif"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="map"
      data-testid="map-div"
    >
      <img
        alt="flat texture blue green World transparent background @transparentpng.com"
        class="world-image"
        src="https://upload.wikimedia.org/wikipedia/commons/a/a9/Rotating_earth_%28large%29_transparent.gif"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing the WorldMap The Worldmap component should match the snapshot:  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="map"
        data-testid="map-div"
      >
        <img
          alt="flat texture blue green World transparent background @transparentpng.com"
          class="world-image"
          src="https://upload.wikimedia.org/wikipedia/commons/a/a9/Rotating_earth_%28large%29_transparent.gif"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="map"
      data-testid="map-div"
    >
      <img
        alt="flat texture blue green World transparent background @transparentpng.com"
        class="world-image"
        src="https://upload.wikimedia.org/wikipedia/commons/a/a9/Rotating_earth_%28large%29_transparent.gif"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing the WorldMap component The WorldMap component should match the snapshot:  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="map"
        data-testid="map-div"
      >
        <img
          alt="flat texture blue green World transparent background @transparentpng.com"
          class="world-image"
          src="https://upload.wikimedia.org/wikipedia/commons/a/a9/Rotating_earth_%28large%29_transparent.gif"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="map"
      data-testid="map-div"
    >
      <img
        alt="flat texture blue green World transparent background @transparentpng.com"
        class="world-image"
        src="https://upload.wikimedia.org/wikipedia/commons/a/a9/Rotating_earth_%28large%29_transparent.gif"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
